### DISCUSSION TITLE
Adding scratch/snap-like visual scripting

### DISCUSSION THREAD

**Discussion by akasob** [godotengine/godot-proposals#5472] [Scripting]
As you know, the visual script will be removed in version 4.
In this regard, I propose to consider the possibility of creating a Scratch-style visual scripting system for Godot.
Here are examples of how it might look in comparison with the old scripting.
Compare how much more concise a block script is than a node script.
![LS1663851680](https://user-images.githubusercontent.com/58305469/191757428-af2c484a-abf7-427a-8b46-6e032d9fd407.png)
![LS1663851663](https://user-images.githubusercontent.com/58305469/191757370-af46dcc5-b0c3-4178-b176-a117e1140621.png)

**Comment by Calinou** [comment]
This is already being tracked in https://github.com/godotengine/godot-visual-script/issues/27 and https://github.com/godotengine/godot-visual-script/issues/29.
